packages/ra-ui-materialui/src/list/SimpleList/SimpleList.spec.tsx

```diff
@@ -143,11 +150,45 @@ describe('<SimpleList />', () => {
                 }}
             >
                 <SimpleList />
-            </ListContext.Provider>
+            </ListContext.Provider>,
+            { wrapper: Wrapper }
         );
         expect(screen.queryByText('ra.navigation.no_results')).not.toBeNull();
     });
 
+    it('should display a message when there is no result but filters applied', async () => {
```

Comment:
Why isn't this test in the FilterButton tests file?

Reply:
Because we test the `<SimpleList>` logic here. I use the `FilterButton`'s story to be able to test it with writing a new component. Do you prefer I write a new component to render that test ?

Reply:
nvm

Reply:
But then, why the SimpleList and not the List? Should we do both? @fzaninotto

Reply:
In fact, the List?oResults is used by `SimpleList` and `Datagrid`, so in theory we should test both. Instead, I added a test for `ListNoResults` itself.